src/conversion.jl

```diff
@@ -92,39 +98,48 @@ end
 for ((TA, fieldA), (TB, fieldB)) in (((MatrixMap, :lmap), (MatrixMap, :lmap)),
                                      ((MatrixMap, :lmap), (UniformScalingMap, :λ)),
                                      ((UniformScalingMap, :λ), (MatrixMap, :lmap)))
-    @eval function Base.convert(::Type{AbstractMatrix}, AB::CompositeMap{<:Any,<:Tuple{$TB,$TA}})
+    @eval function Base.convert(::Type{AbstractMatrix},
+                                AB::CompositeMap{<:Any,<:Tuple{$TB,$TA}})
         B, A = AB.maps
         return A.$fieldA*B.$fieldB
     end
 end
-function Base.Matrix{T}(AB::CompositeMap{<:Any,<:Tuple{MatrixMap,MatrixMap}}) where {T}
+function Base.Matrix{T}(AB::CompositeMap{<:Any, <:Tuple{MatrixMap, MatrixMap}}) where {T}
     B, A = AB.maps
     return convert(Matrix{T}, A.lmap*B.lmap)
 end
-function SparseArrays.sparse(AB::CompositeMap{<:Any,<:Tuple{MatrixMap,MatrixMap}})
+function SparseArrays.sparse(AB::CompositeMap{<:Any, <:Tuple{MatrixMap, MatrixMap}})
     B, A = AB.maps
     return convert(SparseMatrixCSC, A.lmap*B.lmap)
 end
-function Base.Matrix{T}(λA::CompositeMap{<:Any,<:Tuple{MatrixMap,UniformScalingMap}}) where {T}
+function Base.Matrix{T}(λA::CompositeMap{<:Any, <:Tuple{MatrixMap, UniformScalingMap}}) where {T}
     A, J = λA.maps
     return convert(Matrix{T}, J.λ*A.lmap)
+    # if you want less allocations: lmul!(J.λ, convert(Matrix{T}, A.lmap))
 end
-function SparseArrays.sparse(λA::CompositeMap{<:Any,<:Tuple{MatrixMap,UniformScalingMap}})
+function SparseArrays.sparse(λA::CompositeMap{<:Any, <:Tuple{MatrixMap, UniformScalingMap}})
     A, J = λA.maps
     return convert(SparseMatrixCSC, J.λ*A.lmap)
+    # probably also works with sparse: lmul!(J.λ, convert(SparseMatrixCSC, A.lmap))
+
 end
-function Base.Matrix{T}(Aλ::CompositeMap{<:Any,<:Tuple{UniformScalingMap,MatrixMap}}) where {T}
+function Base.Matrix{T}(Aλ::CompositeMap{<:Any, <:Tuple{UniformScalingMap, MatrixMap}}) where {T}
     J, A = Aλ.maps
     return convert(Matrix{T}, A.lmap*J.λ)
+    # if you want less allocations: rmul!(convert(Matrix{T}, A.lmap), J.λ)
 end
-function SparseArrays.sparse(Aλ::CompositeMap{<:Any,<:Tuple{UniformScalingMap,MatrixMap}})
+function SparseArrays.sparse(Aλ::CompositeMap{<:Any, <:Tuple{UniformScalingMap, MatrixMap}})
     J, A = Aλ.maps
     return convert(SparseMatrixCSC, A.lmap*J.λ)
+    # probably also works with sparse: rmul!(convert(SparseMatrixCSC, A.lmap), J.λ)
 end
 
 # BlockMap & BlockDiagonalMap
 Base.Matrix{T}(A::BlockMap) where {T} = hvcat(A.rows, convert.(Matrix{T}, A.maps)...)
 Base.convert(::Type{AbstractMatrix}, A::BlockMap) = hvcat(A.rows, convert.(AbstractMatrix, A.maps)...)
+# seems like code duplication:
+# convert(AbstractMatrix, A) will anyway fall back to Matrix{eltype(A)}(A)
+
```

Comment:
True. But conversion to `AbstractMatrix` keeps special matrix types, where `Matrix` makes them dense.

Reply:
Ok, now I see. I was tired last night :-).

Reply:
Don't worry. Sometimes I'm tired during the day. 😂